lib/src/Uncertainty/Algorithm/Sensitivity/openturns/RankSobolSensitivityAlgorithm.hxx

```diff
@@ -0,0 +1,134 @@
+  Graph draw() const override;
+  
+  /** Design accessor */
+  void setDesign(const Sample & inputDesign,
+                 const Sample & outputDesign);
+                                 
+  /** First order indices accessor */
+  Point getFirstOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** String converter */
+  String __repr__() const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;  
+  
+  /** Second order indices accessor */
+  SymmetricMatrix getSecondOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** Total order indices accessor */
+  Point getTotalOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+
+  /** Interval for the total order indices accessor */
+  Interval getTotalOrderIndicesInterval() const override;
+  
+  /** Aggregated total order indices accessor for multivariate samples */
+  Point getAggregatedTotalOrderIndices() const override;
+  
+  /** Whether to use bootstrap or asymptotic distribution */
+  void setUseAsymptoticDistribution(Bool ) override {};
+  Bool getUseAsymptoticDistribution() const override;
+
+  /** Estimator distribution accessor */
+  Distribution getTotalOrderIndicesDistribution() const override;
+   
+protected:
+
+  Sample computeIndices() const;
+  
+  /** void method that computes bootstrap confidence interval */
+  void computeBootstrapDistribution() const override;
+
+private:
+
+  static Graph DrawSobolFirstIndices(const Description & inputDescription,
+                                     const Point & firstOrderIndices,
+                                     const Interval & firstOrderConfidenceInterval = Interval());
+
+  /** Internal method that compute Vi/VTi using a sample */
+  Sample computeIndices(const Sample & sample, Sample & VTi) const override;
+  
+  Sample computeIndicesFromSample(const Sample &, const Sample &) const;
+  
+  /** Function that computes aggregated indices using Vi/VTi + variance  */
+  Point computeAggregatedIndices(const Sample & Vi,
+                                 const Point & variance) const;
+                                                        
+                                
+  void setDesign(const Sample & ,
+                 const Sample & ,
+                 const UnsignedInteger ) override {};
+                               
+  /** void method that computes asymptotic distribution */
+  void computeAsymptoticDistribution() const override {};
+
+  /** Aggregated first order indices accessor for multivariate samples */
+  Point getAggregatedFirstOrderIndices() const override;
+
+  /** Method that draw  the sensitivity graph of a fixed marginal */
+  Graph draw(const UnsignedInteger marginalIndex) const override;
+      
+  /** Designs : input & output designs */
+  Sample inputDesign_;
```

Comment:
Maybe it is better to set this attribut at `SobolIndicesAlgorithmImplementation` level?